src/public/modules/core/services/gtag.client.service.js

```diff
@@ -1,10 +1,18 @@
+const get = require('lodash/get')
+
 angular.module('core').factory('GTag', ['$rootScope', '$window', GTag])
 
 function GTag($rootScope, $window) {
   // Google Analytics tracking ID provided on signup.
   const GATrackingID = $window.GATrackingID
   let gtagService = {}
 
+  const getUserEmail = () => {
+    const userDetails = JSON.parse($window.localStorage.getItem('user'))
```

Comment:
Sorry I didn't notice this before, but I'm not comfortable with passing the item from local storage straight into `JSON.parse`. If `localStorage.getItem('user')` is anything other than a parsable string, or for that matter even if it's an empty string, `JSON.parse` will throw an error. Can we encase this in a try-catch block to protect against such exceptions? Apologies again for not noticing this before.

Reply:
as discussed on Slack, you can just reuse the `getUser` function from `auth.client.service`

Reply:
thanks! fair point, I've made the edits